build.sbt

```diff
@@ -39,17 +39,21 @@ sources in Test := {
   }
 }
 
-crossScalaVersions := Seq("2.10.7", "2.11.12", "2.12.6", "2.13.0-M3")
+crossScalaVersions := Seq("2.10.7", "2.11.12", "2.12.6", "2.13.0-M4")
 
 javacOptions ++= Seq("-source", "1.7", "-target", "1.7")
 
 scalacOptions ++= Seq(
   "-deprecation",
   "-unchecked",
-  "-Xfuture",
-  "-Yno-adapted-args"
+  "-Xfuture"
 )
 
+scalacOptions ++= PartialFunction.condOpt(CrossVersion.partialVersion(scalaVersion.value)) {
+  case Some((2, v)) if v <= 12 =>
+    "-Yno-adapted-args"
```

Comment:
I think this can actually just be fully removed. We can rely on the build process to fail on compiler versions >= 2.12 to catch adapted args issues if they are introduced.